### DISCUSSION TITLE
Fonts not showing up correctly on arch linux

### DISCUSSION THREAD

**Discussion by socketwiz** [starship/starship#4190] [Q&A] [answered]
I know this is not a bug in Starship, this is a problem with my configuration, but I'm not sure where to go so I thought I would start here.  If there is a better place to take this problem, please let me know.  I have 2 environments, pop os and arch, it works great on pop, not so much on arch.  Here is what I've tried.

1) Install the Kitty terminal.  There is a terminal compatibility list at the bottom of https://github.com/tonsky/FiraCode that suggests Kitty is supported.
2) I chose the "FiraCode Nerd Font" from https://www.nerdfonts.com/font-downloads and installed it
```shell
$ unzip FiraCode.zip -d fira-code
$ cd fira-code
$ sudo cp *.ttf /usr/share/fonts/TTF/
$ cd /usr/share/fonts/TTF/
$ sudo fc-cache -fv .
```
3) I configured Kitty to use the "FiraCode Nerd Font"
```shell
$ kitty list-fonts | grep FiraCode
FiraCode Nerd Font
FiraCode Nerd Font Mono
FiraCode NF
```
I chose `FiraCode Nerd Font`
```
# ~/.config/kitty/kitty.conf
font_family      FiraCode Nerd Font
bold_font        auto
italic_font      auto
bold_italic_font auto

font_size 18.0
```
4) I checked the locale
```shell
$ locale -a
C
C.UTF-8
en_US.utf8
POSIX
```
5) I checked `$TERM`
```shell
$ echo $TERM
xterm-kitty
```
6) I tried rebooting the machine

Here is what I'm getting

![starship-fonts](https://user-images.githubusercontent.com/718481/179961638-6a5c6adf-3f5a-4f5d-bf61-4b43d8cb853f.png)

These are all things that I've found either from the Starship documentation or random google searches trying to solve the issue.  I could use another set of eyes at this point as everything "looks" correct, but clearly I'm missing something.  Thank you in advance for any suggestions you might have.

**Comment by davidkna** [comment, marked as answer]
Please try installing an emoji font (e.g. noto emoji). Alternatively, the [nerd font preset](https://starship.rs/presets/nerd-font.html) reduces the use of emoji greatly.

**Reply by socketwiz** [reply]
Thanks so much for your help. I found the `noto-fonts-emoji` arch package and installed it, but that in and of itself didn't help much.  In my research I ran across this [post](https://dev.to/darksmile92/get-emojis-working-on-arch-linux-with-noto-fonts-emoji-2a9) which contains a script (that I'll post below) that did seem to fix it. I'm not sure if it is the configuration file it created or just flushing the cache. Either way I'm back in business!

Fonts working again!
![starship-fonts-fixed](https://user-images.githubusercontent.com/718481/180034886-72dd882e-a973-4841-aa9b-85571d6b6158.png)

```bash
#!/bin/sh
set -e
if [[ $(id -u) -ne 0 ]] ; then echo "Please run as root" ; exit 1 ; fi
echo "Setting up Noto Emoji font..."
# 1 - install  noto-fonts-emoji package
pacman -S noto-fonts-emoji --needed
# pacman -S powerline-fonts --needed
echo "Recommended system font: inconsolata regular (ttf-inconsolata or powerline-fonts)"
# 2 - add font config to /etc/fonts/conf.d/01-notosans.conf
echo "<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
 <alias>
   <family>sans-serif</family>
   <prefer>
     <family>Noto Sans</family>
     <family>Noto Color Emoji</family>
     <family>Noto Emoji</family>
     <family>DejaVu Sans</family>
   </prefer> 
 </alias>

 <alias>
   <family>serif</family>
   <prefer>
     <family>Noto Serif</family>
     <family>Noto Color Emoji</family>
     <family>Noto Emoji</family>
     <family>DejaVu Serif</family>
   </prefer>
 </alias>

 <alias>
  <family>monospace</family>
  <prefer>
    <family>Noto Mono</family>
    <family>Noto Color Emoji</family>
    <family>Noto Emoji</family>
    <family>DejaVu Sans Mono</family>
   </prefer>
 </alias>
</fontconfig>

" > /etc/fonts/local.conf
# 3 - update font cache via fc-cache
fc-cache
echo "Noto Emoji Font installed! You may need to restart applications like chrome. If chrome displays no symbols or no letters, your default font contains emojis."
echo "consider inconsolata regular"
```
